File: JavaScript/bf.js
const fs = require('fs');

let data = new Array(30000).fill(0);
let p = 0;
let s = "";

function process_brainfuck(loc) {
  let t = loc;
  while (t < s.length) {
    let c = s[t];
    if (c == '+') { data[p]++; }
    if (c == '-') { data[p]--; }
    if (c == '>') { p++; }
    if (c == '<') { p--; }
    if (c == '.') { process.stdout.write(String.fromCharCode(data[p])); }
    if (c == '[') { t = process_brainfuck(t + 1); }
    if (c == ']' && data[p] != 0) {
      if (data[p] != 0) return process_brainfuck(loc);
      return t;
    }
    t++;
  }
  return s.length;
}

const filePath = process.argv[2];
fs.readFile(filePath, 'utf8', (err, content) => {
  if (err) {
    console.error(err);
    return;
  }
  s = content;
  process_brainfuck(0);
});

File: C#/bf.cs
using System;
using System.IO;

class bf {
    static int[] data = new int[30000];
    static int p = 0;
    static string s = "";

    static void Main(string[] args) {
        string filePath = args[0];
        string content = File.ReadAllText(filePath);
        s = content;
        ProcessBrainfuck(0);
    }

    static int ProcessBrainfuck(int loc) {
        int t = loc;
        while (t < s.Length) {
            char c = s[t];
            if (c == '+') data[p]++;
            if (c == '-') data[p]--;
            if (c == '>') p++;
            if (c == '<') p--;
            if (c == '.') Console.Write(Convert.ToChar(data[p]));
            if (c == '[') t = ProcessBrainfuck(t + 1);
            if (c == ']' && data[p] != 0) {
                if (data[p] != 0) return ProcessBrainfuck(loc);
                return t;
            }
            t++;
        }
        return s.Length;
    }
}

